app/controllers/api/v2/providers_controller.rb

```diff
@@ -18,11 +18,29 @@ def show
         render jsonapi: provider, include: params[:include]
       end
 
+      def sync_courses_with_search_and_compare
+        provider = Provider.find_by!(provider_code: params[:code].upcase)
```

Comment:
This has to be changed for our new recruitment-cycle world.

Reply:
I've actioned this and added tests. I've also made the error messaging better like Sau did for the previous PR syncing a single course.